spec/chore/queues/sqs/consumer_spec.rb

```diff
@@ -1,145 +1,194 @@
-require 'spec_helper'
-
-describe Chore::Queues::SQS::Consumer do
-  let(:queue_name) { "test" }
-  let(:queue_url) { "test_url" }
-  let(:queues) { double("queues") }
-  let(:queue) { double("test_queue", :visibility_timeout=>10, :url=>"test_queue", :name=>"test_queue") }
-  let(:options) { {} }
-  let(:consumer) { Chore::Queues::SQS::Consumer.new(queue_name) }
-  let(:message) { TestMessage.new("handle",queue, "message body", 1) }
-  let(:message_data) {{:id=>message.id, :queue=>message.queue.url, :visibility_timeout=>message.queue.visibility_timeout}}
-  let(:pool) { double("pool") }
-  let(:sqs) { double('AWS::SQS') }
-  let(:backoff_func) { nil }
-
-  before do
-    allow(AWS::SQS).to receive(:new).and_return(sqs)
-    allow(sqs).to receive(:queues) { queues }
-
-    allow(queues).to receive(:url_for) { queue_url }
-    allow(queues).to receive(:[]) { queue }
-    allow(queue).to receive(:receive_message) { message }
-    allow(pool).to receive(:empty!) { nil }
-  end
-
-  describe "consuming messages" do
-    let!(:consumer_run_for_one_message) { allow(consumer).to receive(:running?).and_return(true, false) }
-    let!(:messages_be_unique) { allow_any_instance_of(Chore::DuplicateDetector).to receive(:found_duplicate?).and_return(false) }
-    let!(:queue_contain_messages) { allow(queue).to receive(:receive_messages).and_return(message) }
-
-    it 'should configure sqs' do
-      allow(Chore.config).to receive(:aws_access_key).and_return('key')
-      allow(Chore.config).to receive(:aws_secret_key).and_return('secret')
-
-      expect(AWS::SQS).to receive(:new).with(
-        :access_key_id => 'key',
-        :secret_access_key => 'secret'
-      ).and_return(sqs)
-      consumer.consume
-    end
-
-    it 'should not configure sqs multiple times' do
-      allow(consumer).to receive(:running?).and_return(true, true, false)
-
-      expect(AWS::SQS).to receive(:new).once.and_return(sqs)
-      consumer.consume
+require_relative '../spec_helper'
+
+module Chore
+  describe Chore::Queues::SQS::Consumer do
+    include_context 'fake objects'
+
+    let(:options) { {} }
+    let(:consumer) { Chore::Queues::SQS::Consumer.new(queue_name) }
+    let(:backoff_func) { Proc.new { 2 + 2 } }
+    let(:dupe_detector) { double(Chore::DuplicateDetector) }
+    let(:empty_message_collection) { double(Aws::SQS::Message::Collection.new([])) }
+
+    let(:receive_message_result) { Aws::SQS::Message::Collection.new([message], size: 1) }
+
+    let(:message) do
+       Aws::SQS::Message.new(
+        message_id: 'message id',
+        receipt_handle: "receipt_handle",
+        body: job.to_json,
+        data: job,
+        queue: queue,
+        queue_url: sqs.get_queue_url.queue_url,
```

Comment:
This seems like we're taking advantage of the fact that the `get_queue_url` method doesn't actually require any arguments which I think isn't actually the case and is a bit confusing.  I think it'd be clearer to reference the `queue_url` variable here.